screens/Scan/ScanLayoutController.ts

```diff
@@ -63,6 +63,10 @@ export function useScanLayout() {
   const onStayInProgress = () =>
     scanService.send(ScanEvents.STAY_IN_PROGRESS());
   const onRetry = () => scanService.send(ScanEvents.RETRY());
+  const GOTO_HISTORY = () => {
+    scanService.send(ScanEvents.GOTO_HISTORY());
```

Comment:
Can we change the design here to send DISCONNECT/CLEANUP event which would be listened by machine & that machine invokes wallet disconnect.

This same event can also be sent in GO_TO_HOME, so in the machine we don't need to handle 2 states for wallet disconnection (LINE 617 - 626 in ScanMachine.ts)

Reply:
@vijay151096 we require your inputs here, lets connect on this?